### DISCUSSION TITLE
Is it possible to get cosmos to only build the components it needs?

### DISCUSSION THREAD

**Discussion by akramscorado** [react-cosmos/react-cosmos#1541] [Q&A]
We are considering using cosmos with our `next.js` app and it seems to works well in the `development environment`. However, to get it to build statically it requires the `whole app` to be built and exported (as the docs suggest).

What we need, is for cosmos to statically build only the components it needs and export them in a separate folder. Is that possible?
Thanks

**Comment by ovidiuch** [comment]
I'd like to figure this out, too. Unfortunataly, as mentioned in the docs, Next.js doesn't seem to have an API for interacting with it as a dev tool. I've tried to reach out to the Next.js community on Twitter and their Discord but no luck.

If anyone has ideas for how to essentially build a single page/segment from a Next.js app I'd love to hear them. I can definitely help make the process smooth on the Cosmos side but at the moment Next.js is kind of a black box.

**Comment by akramscorado** [comment]
Hi again. 

We really liked react-cosmos so we tried to get it to run in a way to satisfy our needs with our NextJS app. What we needed was to have a self-contained instance of cosmos running independently capable of static independent deployment outside the app without the need to `statically build` the whole app. So this is our solution:
* We instantiated react-cosmos within a designated subfolder residing within our application's structure and we called it `.cosmos`.
* This sub-app runs NextJS 13 with the bare minimum requirement. Its main job is to have a list of `fixtures`.
* The `fixtures` can access the main-app's components by configuring nextJS to allow access to `externalDir`
* The app/cosmos/[fixture]/page.tsx is then added to the main app.

Now, `.cosmos` can run/build independently without the need to run the main app itself.

We have also created an repo example for our solution:
https://github.com/scorado-ltd/scorado-examples-nextjs-cosmos/tree/main

Thanks

**Comment by ovidiuch** [comment]
@akramscorado That's awesome, thanks for creating the example repo! Feel free to add a link to it in the Next.js docs.
